state/factory/workingset.go

```diff
@@ -34,6 +35,8 @@ var (
 		},
 		[]string{"type"},
 	)
+
+	errUnsupportWeb3Rewarding = errors.New("unsupported web3 rewarding")
```

Comment:
`errors.New` will fill unnecessary stack info in the global defined `errUnsupportWeb3Rewarding `. `fmt.Errorf` maybe better ?

Reply:
1. we used `errors.New` a lot in the codebase, so I think OK to use it to be consistent
2. if we want, we can study this and file a separate PR to replace with `fmt.Errorf`

Reply:
sure, the consistency is also important. we can study it in a separate issue.